bugsnag-plugin-android-anr/src/main/jni/anr_handler.c

```diff
@@ -131,34 +229,16 @@ static bool configure_anr_jni_impl(JNIEnv *env) {
       "(Ljava/lang/String;Ljava/lang/String;Ljava/lang/Number;Ljava/lang/"
       "Long;Ljava/lang/Long;Ljava/lang/Long;Ljava/lang/Boolean;Lcom/bugsnag/"
       "android/ErrorType;)V");
-  return true;
-}
-
-static void safe_delete_local_ref(JNIEnv *env, jobject obj) {
-  if (env != NULL) {
-    (*env)->DeleteLocalRef(env, obj);
+  if (frame_init == NULL) {
+    return false;
   }
-}
 
-static jobject safe_new_object(JNIEnv *env, jclass clz, jmethodID method, ...) {
-  if (env == NULL || clz == NULL) {
-    return NULL;
+  if ((*env)->GetJavaVM(env, &jvm) != JNI_OK) {
```

Comment:
I put it at the end so that `jvm` is always null if anything went wrong during initialization, and thus no other function will run since they all check. Since this is initialization code, it doesn't matter if it does useless work in a failure case.